stock_move_auto_assign/models/product_product.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import models
+from odoo.addons.queue_job.job import job
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    def _moves_auto_assign_domain(self, locations):
+        return [
+            ("product_id", "=", self.id),
+            ("location_id", "parent_of", locations.ids),
+            ("state", "in", ("confirmed", "partially_available")),
+            # useless to try reserving a move that waits on another move
+            # or is MTO
+            ("move_orig_ids", "=", False),
+            ("procure_method", "=", "make_to_stock"),
+            # Do not filter on product_id.type by default because it uses an
+            # additional query on product_product and product_template.
+            # StockMove._prepare_auto_assign() already filtered out
+            # non-stockable products and # `_action_assign()` would filter them
+            # out anyway.
+        ]
```

Comment:
Wouldn't be good to add a date (date_expected) in this domain? This is easily extensible, but anyway you wouldn't want to reserve inventory for moves that are planned to be done next month for instance.

Reply:
@LoisRForgeFlow Odoo standard behavior is actually to reserve goods at each scheduler run whatever their dates. This module doesn't intend to change this principle, only avoid to launch the scheduler every 10 minutes.

If you want to avoid move reserving goods a month in advance, you shall use our excellent module :) https://github.com/OCA/wms/tree/13.0/sale_stock_available_to_promise_release 

This will allow you to chose what you release and intend the system to reserve or not.

With this module, every move within the warehouse is supposed to be treated and therefor reserved.

Reply:
Fair enough, thanks for the insights. Is your recommendation to reduce the frequency of the scheduler when using this module?

Reply:
Yes, of course, as then you probably don't need it anymore for short term, only once a day shall be enough IMO